esmvalcore/local.py

```diff
@@ -467,15 +520,33 @@ def _select_drs(input_type: str, project: str, structure: str) -> list[str]:
 
 
 @dataclass(order=True)
-class DataSource:
-    """Class for storing a data source and finding the associated files."""
+class LocalDataSource(esmvalcore.io.protocol.DataSource):
+    """Data source for finding files on a local filesystem."""
+
+    name: str
+    """A name identifying the data source."""
+
+    project: str
+    """The project that the data source provides data for."""
+
+    priority: int
+    """The priority of the data source. Lower values have priority."""
```

Comment:
project is well documented, priority is not at all. I myself struggle to understand the functionality behind it, it'd be great if you could document it, since I see it in the new configurations for various datasets. Also, the phrase "The priority of the data source. Lower values have priority." (which is repeated in various modules) is as vague as a foggy morning

Reply:
cheers, that's what I gathered too - but it'd be nice to document it, since it's something that the user has *some* control over; also remember that `priority` is also assigned to a `preprocessor.Task` - totally different variable, same name - just so we don't mix jars. Can we use (this) priority to search different ESGF nodes preferentially?

Reply:
I've deprecated `search_esgf` and introduced a new `search_data` to replace it now

Reply:
There is now a detailed explanation of `priority` in https://esmvaltool--2765.org.readthedocs.build/projects/ESMValCore/en/2765/api/esmvalcore.io.html and a short one in https://esmvaltool--2765.org.readthedocs.build/projects/ESMValCore/en/2765/quickstart/configure.html#top-level-configuration-options